core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -231,22 +244,39 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
                 }
                 return fields;
             }
+
+            private void validateParsed(Float value) {
+                if (
+                    value.isNaN() || value.isInfinite()
+                        || value > 65504
```

Comment:
should probably be `Math.abs(value) >= 65520` rather than 65504. 65504 is indeed the maximum value but values up to 65520 excluded would be rounded to 65504